### PR TITLE
ast: add `create role` ast structure

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
add `CREATE ROLE` support by using `CreateUserStmt`.

### What is changed and how it works?
add boolean variable `IsCreateRole` to distinct `CREATE ROLE` with `CREATE USER`. 

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

Related changes

### CODE REVIEW

File: go.mod1 [hunk 1]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/pingcap/errors v0.11.0
 	github.com/pingcap/tidb v0.0.0-20190218065808-69472bd1a6e9
 	github.com/pingcap/tipb v0.0.0-20190107072121-abbec73437b7
+	github.com/pkg/errors v0.8.1
```
Comment (kennytm) [inline go.mod1]: cc @tiancaiamao Why is everyone adding this line to `go.mod1`?

File: parser.y [hunk 2]
```diff
@@ -7396,6 +7396,7 @@ CreateUserStmt:
 	{
  		// See https://dev.mysql.com/doc/refman/5.7/en/create-user.html
 		$$ = &ast.CreateUserStmt{
+		    IsCreateRole: false,
```
Comment (kennytm) [inline parser.y]:
```suggestion
			IsCreateRole: false,
```

File: go.mod1 (hunk 1 above)
Comment (amyangfei) [reply go.mod1]: I find this commit https://github.com/pingcap/parser/commit/60fd134cfde20424ba588c785debb3be0a086210 introduced the usage `github.com/pkg/errors`. Should we replace it with `github.com/pingcap/errors`?

**Comment by tiancaiamao** [comment]
Is there a TiDB PR? @imtbkcat

**Comment by imtbkcat** [comment]
@tiancaiamao yes, here is : https://github.com/pingcap/tidb/pull/9461

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**